README.md

```diff
@@ -34,3 +34,205 @@ Read so far up to
 - Install [bun](https://bun.sh/)
 - Run `bun install` to install all dependencies
 - Run `bun test` to run all tests
+
+## Introduction to Cairo VM
+Cairo VM is an advanced virtual CPU architecture designed for blockchain technology and cryptographic operations. It's Turing-complete and STARK-friendly, efficiently handling complex program statements as polynomial equations over a finite field. This feature is crucial for secure and effective cryptographic proof management in blockchain applications, making Cairo VM a key player in the realm of decentralized computing.
+
+## Register-Based VM
+Cairo VM's register-based architecture allows for quicker data access compared to stack based interaction, enhancing computational efficiency and performance in cryptographic operations.
+
+## Detailed Architecture Overview
+- **Finite Field Operations**: Core to Cairo VM's computational power for cryptographic aspects of STARKs.
+
+- **Three-Register System**: The Cairo VM employs a three-register system, consisting of:
+
+    - **Program Counter (PC)**: Manages the sequence of program execution by holding the memory address of the next instruction.
+    - **Allocation Pointer (AP)**: Crucial for memory management, it points to the first unused memory cell, dynamically adjusting as the program runs.
```

Comment:
plz avoid words like "crucial" and in general, words that don't give a precise and concise info

Reply:
"dynamically adjusting" ?

Reply:
That it changes not linearly but "dynamically" as the program runs